src/suspendplugin.c

```diff
@@ -407,19 +460,16 @@ suspend_plugin_init()
         mce_display_add_state_changed_handler(mce_display,
             suspend_display_event, NULL);
 
-    nl_socket = nl_socket_alloc();
+    nl_socket_fd = nl_socket_get_fd(nl_socket);
+    netlink_channel = g_io_channel_unix_new(nl_socket_fd);
+    g_io_channel_set_close_on_unref(netlink_channel, TRUE);
 
-    genl_connect(nl_socket);
+    g_io_channel_set_encoding(netlink_channel, NULL, NULL);
+    g_io_channel_set_buffered(netlink_channel, FALSE);
```

Comment:
Not sure if this encoding/buffering is needed as glib io routines are not used for reading.